lib/src/transformer/transformer.dart

```diff
@@ -14,7 +14,12 @@ import '../asset/writer.dart';
 import '../builder/builder.dart';
 import '../builder/build_step_impl.dart';
 
-abstract class BuilderTransformer implements Transformer, DeclaringTransformer {
+/// A [Transformer] which runs multiple [Builder]s.
+/// Extend this class and define the [builders] getter to create a [Transformer]
```

Comment:
I still think we should be explicit about our (lack of) ordering guarantees here, maybe just as simple as "there are no guarantees about the order in which [Builder]s will be run."